File: tests/fixtures/ast-treesitter/sample.ts
import { readFile } from 'node:fs/promises'
import * as path from 'node:path'
import defaultExport from './helpers'

export const GREETING = 'hello'
export let counter = 0

export function greet(name: string): string {
  return `${GREETING}, ${name}`
}

export class Logger {
  log(msg: string) {
    console.log(msg)
  }
}

export interface Config {
  verbose: boolean
}

export type Handler = (x: number) => number

export default function main() {
  return readFile(path.join('.', 'x'))
}

File: tests/fixtures/ast-treesitter/sample.py
import os
import sys
from pathlib import Path
from typing import List, Optional

GREETING = "hello"
_private = 42

def greet(name: str) -> str:
    return f"{GREETING}, {name}"

async def fetch_config(path: str) -> Optional[str]:
    return None

class Logger:
    def __init__(self):
        self.entries: List[str] = []

    def log(self, msg: str) -> None:
        self.entries.append(msg)

class _Hidden:
    pass
